stan/fit.py

```diff
@@ -128,12 +128,16 @@ def __getitem__(self, param):
         assert self._draws.shape == (len(self.sample_and_sampler_param_names) + len(self.constrained_param_names), num_samples_saved, self.num_chains)
         # fmt: on
         # Stack chains together. Parameter is still stored flat.
-        view = self._draws[param_indexes, :, :].reshape(len(param_indexes), -1).view()
-        assert view.shape == (len(param_indexes), num_samples_saved * self.num_chains)
-        # reshape must yield something with least two dimensions
-        reshape_args = param_dim + [-1] if param_dim else (1, -1)
-        # reshape, recover the shape of the stan parameter
-        return view.reshape(*reshape_args, order="F")
+        if len(param_indexes):
+            view = self._draws[param_indexes, :, :].reshape(len(param_indexes), -1).view()
+            assert view.shape == (len(param_indexes), num_samples_saved * self.num_chains)
+            # reshape must yield something with least two dimensions
+            reshape_args = param_dim + [-1] if param_dim else (1, -1)
+            # reshape, recover the shape of the stan parameter
+            view = view.reshape(*reshape_args, order="F")
+        else:
+            view = np.empty(param_dim + [num_samples_saved * self.num_chains])
+        return view
```

Comment:
```suggestion
        if not len(param_indexes):
            return np.empty(param_dim + [num_samples_saved * self.num_chains])
        view = self._draws[param_indexes, :, :].reshape(len(param_indexes), -1).view()
        assert view.shape == (len(param_indexes), num_samples_saved * self.num_chains)
        # reshape must yield something with least two dimensions
        reshape_args = param_dim + [-1] if param_dim else (1, -1)
        # reshape, recover the shape of the stan parameter
        return view.reshape(*reshape_args, order="F")
```
Two fewer lines and easier to read, I think.

I might add a comment right after the `if` statement and explain that a zero-length parameter is involved.

Also, Surely the `np.empty` is wrong? That's just going to be random data (uninitialized memory).

Reply:
I think `np.empty` is fine, it is 0-dimensional object containing nothing.